test/test.js

```diff
@@ -1,27 +1,50 @@
 function readBlobAsText(blob) {
   return new Promise(function(resolve, reject) {
-    var reader = new FileReader()
-    reader.onload = function() {
-      resolve(reader.result)
-    }
-    reader.onerror = function() {
-      reject(reader.error)
+    // FileReader is not available on Workers in Firefox, but FileReaderSync
+    // is.
+    var reader
+    if (typeof FileReaderSync !== "undefined") {
```

Comment:
shouldn't you check for FileReader first instead, to use the async version when both are available ?